presto-main/src/test/java/com/facebook/presto/type/TestCharOperators.java

```diff
@@ -0,0 +1,108 @@
+        assertFunction("cast('foo' as char(3)) = cast('bar' as char(3))", BOOLEAN, false);
+        assertFunction("cast('bar' as char(3)) = cast('foo' as char(3))", BOOLEAN, false);
+    }
+
+    @Test
+    public void testNotEqual()
+            throws Exception
+    {
+        assertFunction("cast('foo' as char(3)) <> cast('foo' as char(3))", BOOLEAN, false);
+        assertFunction("cast('foo  ' as char(5)) <> cast('foo' as char(5))", BOOLEAN, false);
+        assertFunction("cast('foo' as char(3)) <> cast('bar' as char(3))", BOOLEAN, true);
+        assertFunction("cast('bar' as char(3)) <> cast('foo' as char(3))", BOOLEAN, true);
+    }
+
+    @Test
+    public void testLessThan()
+            throws Exception
+    {
+        assertFunction("cast('foo' as char(3)) < cast('foo' as char(3))", BOOLEAN, false);
+        assertFunction("cast('foo' as char(3)) < cast('foo' as char(5))", BOOLEAN, true);
+        assertFunction("cast('foo' as char(5)) < cast('foo' as char(3))", BOOLEAN, false);
+        assertFunction("cast('foo' as char(3)) < cast('bar' as char(3))", BOOLEAN, false);
+        assertFunction("cast('bar' as char(3)) < cast('foo' as char(3))", BOOLEAN, true);
+    }
+
+    @Test
+    public void testLessThanOrEqual()
+            throws Exception
+    {
+        assertFunction("cast('foo' as char(3)) <= cast('foo' as char(3))", BOOLEAN, true);
+        assertFunction("cast('foo' as char(3)) <= cast('foo' as char(5))", BOOLEAN, true);
+        assertFunction("cast('foo' as char(5)) <= cast('foo' as char(3))", BOOLEAN, false);
+        assertFunction("cast('foo' as char(3)) <= cast('bar' as char(3))", BOOLEAN, false);
+        assertFunction("cast('bar' as char(3)) <= cast('foo' as char(3))", BOOLEAN, true);
+    }
+
+    @Test
+    public void testGreaterThan()
+            throws Exception
+    {
+        assertFunction("cast('foo' as char(3)) > cast('foo' as char(3))", BOOLEAN, false);
+        assertFunction("cast('foo' as char(3)) > cast('foo' as char(5))", BOOLEAN, false);
+        assertFunction("cast('foo' as char(5)) > cast('foo' as char(3))", BOOLEAN, true);
+        assertFunction("cast('foo' as char(3)) > cast('bar' as char(3))", BOOLEAN, true);
+        assertFunction("cast('bar' as char(3)) > cast('foo' as char(3))", BOOLEAN, false);
+    }
+
+    @Test
+    public void testGreaterThanOrEqual()
+            throws Exception
+    {
+        assertFunction("cast('foo' as char(3)) >= cast('foo' as char(3))", BOOLEAN, true);
+        assertFunction("cast('foo' as char(3)) >= cast('foo' as char(5))", BOOLEAN, false);
+        assertFunction("cast('foo' as char(5)) >= cast('foo' as char(3))", BOOLEAN, true);
+        assertFunction("cast('foo' as char(3)) >= cast('bar' as char(3))", BOOLEAN, true);
+        assertFunction("cast('bar' as char(3)) >= cast('foo' as char(3))", BOOLEAN, false);
+    }
+
+    @Test
+    public void testBetween()
```

Comment:
added comment for length based comparison to distinguish them. Other are just plain internal tests